util/topgen/intermodule.py

```diff
@@ -390,6 +390,38 @@ def check_intermodule_field(obj: OrderedDict, prefix: str = "") -> int:
     return error
 
 
+def find_otherside_modules(topcfg: OrderedDict, m, s) -> List[Tuple[str, str]]:
+    """Find far-end port based on given module and signal name
+    """
+    signame = "{}.{}".format(m, s)
+    for req, rsps in topcfg["inter_module"]["connect"].items():
+        if req.startswith(signame):
+            # return rsps after splitting module instance name and the port
+            result = []
+            for rsp in rsps:
+                rsp_m, rsp_s, rsp_i = filter_index(rsp)
+                result.append((rsp_m, rsp_s))
+            return result
+
+        for rsp in rsps:
+            if signame == rsp:
+                req_m, req_s, req_i = filter_index(req)
+                return [(req_m, req_s)]
+
+    # If reaches here, no matching results in 'connect'
+    # so search 'top' or 'external'
+
+    # todo: something here
+    for sig in topcfg["inter_signal"]["signals"]:
+        if sig["inst_name"] == m and sig["name"] == s:
+            return [("", sig["top_signame"])]
```

Comment:
this doesn't cover the `external` case for now right? i also wonder if you should return another element in the tuple to indicate whether the far side is another module, top or external (not sure if that information is useful)

Reply:
That's good idea. Let me revise the function to return that information too.

Reply:
FYI, `top_signame` also covers the external case too.

Reply:
as discussed, `top_signame` is still interconnect net name. We need to have far end IP instance and IP port name.